src/Task/Filesystem/CopyDir.php

```diff
@@ -113,14 +114,15 @@ protected function copyDir($src, $dst)
             mkdir($dst, $this->chmod, true);
         }
         while (false !== ($file = readdir($dir))) {
-            if (in_array($file, $this->exclude)) {
-                 continue;
+            // Support basename and full path exclusion.
+            if (in_array($file, $this->exclude) || in_array($parent . $file, $this->exclude) || in_array($src . DIRECTORY_SEPARATOR . $file, $this->exclude)) {
```

Comment:
This seems to disallow exclusions such as `nested3/nested31/structu311.re` -- i.e. only the full relative path, or one directory + current file / dir is checked. Is this intended?

If this was intended, I suppose it is fine, as long as it is documented.